### DISCUSSION TITLE
Where in scons do you tell it to find the include (.h) files?

### DISCUSSION THREAD

**Discussion by DerrickLau** [RT-Thread/rt-thread#7916] [Q&A] [answered]
Hello again:

On a separate note, I discovered that when I was trying to build the raspberry pi 4 64bit project in the main branch, one of the .c files could not include the <resource_id.h> file.  When I changed the #include to reference the full directory path and filename, it worked.

Where in scons do I tell the compiler where to find the include files?

Thanks,
Derrick

**Comment by supperthomas** [comment]
we have use ci to build the bsp:

https://github.com/RT-Thread/rt-thread/actions/runs/5769565836/job/15641722973

It seems  ok.

you can follow the CI step to build your bsp.

![image](https://github.com/RT-Thread/rt-thread/assets/60349489/dc664220-8c86-4cb5-a1ad-7debd2e706d7)

**Comment by DerrickLau** [comment]
I will point out that the CI build was not done from the master branch (ie the trunk).  You built from a different branch.  Is it the rt-thread branch that the CI build ran against?  I will grab that branch and I bet you I will be able to build it exactly like you did.

**Reply by supperthomas** [reply, marked as answer]
The CI is the master branch which is the lastest code. I just run the CI after you ask the question. You can check the ci step with your step which is different.

**Reply by DerrickLau** [reply]
Thanks.  I checked the CI steps and found the issue.  You used the toolchain you have stored in github: https://github.com/RT-Thread/toolchains-ci/releases.  The CI steps also showed you built on Ubuntu.  I didn't.  I built on Windows and used this toolchain:  https://developer.arm.com/-/media/Files/downloads/gnu/12.3.rel1/binrel/arm-gnu-toolchain-12.3.rel1-mingw-w64-i686-aarch64-none-elf.zip?rev=b9e2d7090d0240cca2dfa9a902666e4c&hash=AED4968D976D424304AF2B940E633A5C
I guess you build on Ubuntu as it is what Raspbian OS is made from?  I am guessing all the drivers for Raspberry Pi are built on Ubuntu?  
I ask because the latest scons error I am getting is:
C:\Derrick\GitHub\rt-thread-stable0\components\libc\posix\io\stdio\libc.c: In function 'libc_stdio_set_console':
C:\Derrick\GitHub\rt-thread-stable0\components\libc\posix\io\stdio\libc.c:124:22: error: 'struct _reent' has no member named '__sdidinit'
  124 |         _GLOBAL_REENT->__sdidinit = 1;
      |                      ^~
scons: *** [build\kernel\components\libc\posix\io\stdio\libc.o] Error 1
scons: building terminated because of errors.

I'm guessing that member struct _reent->_sdidinit is an ubuntu thing...I suspect my Windows based toolchain won't have that.

**Reply by supperthomas** [reply]
I think you can use the same toolchain (windows type) in windows will be ok.

**Comment by DerrickLau** [comment]
Closing this discussion as a Windows specific issue.

**Reply by supperthomas** [reply]
Thank you for report. We will check the windows environment later.

**Comment by supperthomas** [comment]
tool chain arm-gnu-toolchain-12.3 not support yet currently. You can use 10.xx version

**Reply by DerrickLau** [reply]
Thank you for pointing out my obvious mistake.  My apologies for sucking up your time and grattitude for your perseverence and dilligence.  Switched to 10.xx toolchain and builds perfectly now.

**Reply by supperthomas** [reply]
It's thanks for you to report this issue . We will fix it by PR like #7958 or fix it in future.

**Reply by supperthomas** [reply]
BTW， It's not your mistake. It is compatible issue.
